pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -566,18 +566,22 @@ func doSafeOpen(pathname string, base string) (int, error) {
 	// Follow the segments one by one using openat() to make
 	// sure the user cannot change already existing directories into symlinks.
 	for _, seg := range segments {
+		var deviceStat unix.Stat_t
+
 		currentPath = filepath.Join(currentPath, seg)
 		if !mount.PathWithinBase(currentPath, base) {
 			return -1, fmt.Errorf("path %s is outside of allowed base %s", currentPath, base)
 		}
 
+		// trigger auto mount, ignore error if not a directory
```

Comment:
Please add a comment that `unix.Fstat("path/a")` below is not enough and extra `path/a/` here is needed to trigger automount in case `a` is an automounted dir.

Reply:
Updated the comment and rebased on latest master branch, please check again, thanks!